### PR TITLE
Add diffCustom param to createPatch

### PR DESCRIPTION
**Pull request by diachedelic**
See #29

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -34,19 +34,33 @@ export function applyPatch(object, patch) {
...
-export function createPatch(input, output): Operation[] {
+export function createPatch(
+  input: any,
+  output: any,
+  diffCustom: IDiffCustom = (() => {}),
+): Operation[] {
   const ptr = new Pointer()
   // a new Pointer gets a default path of [''] if not specified
-  return diffAny(input, output, ptr)
+  return diffAny(input, output, ptr, (input, output, ptr) => {
```
Comment (chbrown) [inline index.ts]: What if user wants to intercept root-most case? I.e., this never calls `diffCustom` on the `input` and `output` values that are fed into `createPatch`.

File: diff.ts [hunk 2]
```diff
@@ -265,26 +279,35 @@ export function diffObjects(input: any, output: any, ptr: Pointer): Operation[]
...
+): Operation[] {
   if (!compare(input, output)) {
     return [{op: 'replace', path: ptr.toString(), value: output}]
   }
   return []
 }
 
-export function diffAny(input: any, output: any, ptr: Pointer): Operation[] {
+export function diffAny(
+  input: any,
```
Comment (chbrown) [inline diff.ts]: Please trim trailing whitespace.

File: tests/issues.ts [hunk 3]
```diff
@@ -101,3 +102,33 @@ describe('issues/12', () => {
     assert.deepEqual(patch_results, [null, null, null])
   })
 })
+
+describe('issues/29', () => {
+  var input = {
```
Comment (chbrown) [inline tests/issues.ts]: Ok, I thought of an edge-case!

Replace these three vars with:
```js
  var input = {
    root: {
      diffed: ['a', 'b'],
      not_diffed: ['a', 'b'],
    },
  }
  var output = {
    root: {
      diffed: ['a'],
      not_diffed: ['a'],
    }
  }
  var expected_patch = [
    {op: 'remove', path: '/root/diffed/1'},
    {op: 'replace', path: '/root/not_diffed', value: ['a']},
  ]
```

This fails, saying:
```patch
+ expected 
- actual

     "op": "remove"
     "path": "/root/diffed/1"
   }
   {
-    "op": "remove"
-    "path": "/root/not_diffed/1"
+    "op": "replace"
+    "path": "/root/not_diffed"
+    "value": [
+      "a"
+    ]
   }
 ]
```
